src/components/filters/DateFilter/utils/OptionUtils.ts

```diff
@@ -1,10 +1,26 @@
 // (C) 2007-2019 GoodData Corporation
+import moment = require("moment");
+import isString = require("lodash/isString");
 import { ExtendedDateFilters } from "@gooddata/typings";
 
+const NUM_OF_MILISECOND_IN_HOUR = 60000;
 export const getDateFilterOptionGranularity = (
     dateFilterOption: ExtendedDateFilters.DateFilterOption,
 ): ExtendedDateFilters.DateFilterGranularity =>
     ExtendedDateFilters.isRelativeDateFilterForm(dateFilterOption) ||
     ExtendedDateFilters.isRelativeDateFilterPreset(dateFilterOption)
         ? dateFilterOption.granularity
         : undefined;
+
+export const areSameDate = (from: Date | string, to: Date | string): boolean =>
+    isString(from) && isString(to) ? from === to : moment(from).isSame(moment(to), "hour");
```

Comment:
using the granularity argument as "hour" may cause the function is false in case two dates are just different on timezone. It should be "day" @lvkhanh